### PR TITLE
make ThemedInput noRight(Left)BorderRadius work

### PR DESCRIPTION
**Pull request by ChenRoth**
### What changed

Added css classes for making `<Input noRightBorderRadius noLeftBorderRadius />` work

### Why it changed

these props didn't work properly

---

- [x] Change is tested
- [x] Change is documented
- [ ] Build is green
- [x] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

File: src/Input/ThemedInput.js [hunk 1]
```diff
@@ -32,11 +32,21 @@ class ThemedInput extends Input {
...
       [styles.roundInput]: roundInput,
-      [styles.hasValue]: (value && value.length) || (this.input && !!this.input.value),
-      [noRightBorderRadius]: noRightBorderRadius,
-      [noLeftBorderRadius]: noLeftBorderRadius
+      [styles.hasValue]: (value && value.length) || (this.input && !!this.input.value)
     };
 
+    if (noRightBorderRadius === true) {
+      classes[styles.noRightBorderRadius] = true;
+    } else if (noRightBorderRadius) {
```
Comment (ChenRoth) [inline src/Input/ThemedInput.js]: i had to preserve some hack which used `noRightBorderRadius` prop as a string to inject a classname

File: src/Input/ThemedInput.js [hunk 2]
```diff
@@ -32,11 +32,21 @@ class ThemedInput extends Input {
       [styles.hasHover]: forceHover,
       [styles.hasFocus]: forceFocus || this.state.focus,
       [styles.roundInput]: roundInput,
-      [styles.hasValue]: (value && value.length) || (this.input && !!this.input.value),
-      [noRightBorderRadius]: noRightBorderRadius,
```
Comment (shlomitc) [inline src/Input/ThemedInput.js]: just do it here
`[noRightBorderRadius]: noRightBorderRadius`
should be:
`[styles.noRightBorderRadius]: noRightBorderRadius`

it should be only boolean and not a string (need to remove from the docs as well)

File: src/Input/ThemedInput.js (hunk 1 above)
Comment (shlomitc) [reply src/Input/ThemedInput.js]: let's remove it

File: src/Input/ThemedInput.js (hunk 2 above)
Comment (ChenRoth) [reply src/Input/ThemedInput.js]: well, it would introduce a breaking chance

Comment (ChenRoth) [reply src/Input/ThemedInput.js]: ok, i understand the motivation for that hack and i'll remove it

**Comment by shlomitc** [comment]
So I compared this PR storybook and the regular WSR and there are some minor changes the range now need several adjustments:

1. InputRange
  1.1. the width of it was changed and the left input outline is overriden by the right input. maybe it's the right behvavior though. @milkyfruit, can you verify?
  1.2. One of the intersecting borders should disappear (it's double)
2. DateRange
  2.1. you actually fixed this one with rounded edges, cool :)
  2.2. same like 1.2


## Before
https://wix-wix-style-react.surge.sh/?selectedKind=3.%20Inputs&selectedStory=3.7%20Range&full=0&addons=0&stories=1&panelRight=0

### InputRange
image
image
image

### DateRange
image
image
image


## After
https://wix-wix-style-react-pr-1794.surge.sh/?selectedKind=3.%20Inputs&selectedStory=3.7%20Range&full=0&addons=0&stories=1&panelRight=0

### Input Range
image
image
image

### Date Range
image
image
image

**Comment by milkyfruit** [comment]
@shlomitc both number and date range inputs must have 1px border line in the middle. I see two ways to achieve it:
A. overlap each over by 1px
B. The second input doesn't show the left border on default state

In both suggested options the active input must have a higher number of z-index, like range input in "before" example:
image

**Comment by hepiyellow** [comment]
I pushed some fixes.
### `<Input>` className
I added className prop to <Input> : We should write in Migration that if someone used noRightBorderRadius as a hack to pass class name to the Input, then they should now use the `className` prop.


### `<DatePicker >`- inputProps
 I also fixed Range with DatePicker which was  broken before.
- I added `inputProps` to DatePicker

I think everything works well now.

Just need to add the migration doc.
I saw that there are 7 occurences of noRightBorderRadius in wix-private where they are passing class names and not boolean (which didn't work).
I didn't verify if some of them are already using WSR 4.
If all of them are NOT using WSR 4, then we can go ahead and release this in 4.1.1, and add this to the migration doc of 4.0.0 (while mentioning it is actually implemented in 4.1.1)

**Comment by hepiyellow** [comment]
Failing tests are e2e snapshots.
1 )is a flaky test of unrelated EditableSelector
2) 2 other are expected fixes

**Pull request merged**